src/sinol_make/commands/inwer/__init__.py

```diff
@@ -53,9 +53,11 @@ def verify_test(execution: InwerExecution) -> VerificationResult:
         output_dir = os.path.join(os.getcwd(), 'cache', 'executions', execution.test_name)
         os.makedirs(output_dir, exist_ok=True)
 
-        command = f'{execution.inwer_exe_path} < {execution.test_path}'
-        process = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        test = open(execution.test_path, 'r')
+        command = [execution.inwer_exe_path]
+        process = subprocess.Popen(command, stdin=test, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
         process.wait()
+        test.close()
```

Comment:
Because of readability and error-proneness, it's preferred to open a file with the `with` statement: https://docs.python.org/3/reference/compound_stmts.html#the-with-statement

Reply:
This comment can be applied to other parts of code in this PR.